src/main/resources/io/viash/platforms/nextflow/VDSL3Helper.nf

```diff
@@ -491,12 +492,86 @@ def processProcessArgs(Map args) {
     processArgs.directives.keySet().removeAll(["publishDir", "cpus", "memory", "label"])
   }
 
-  for (nam in [ "map", "mapId", "mapData", "mapPassthrough", "filter" ]) {
+  for (nam in [ "map", "mapId", "mapData", "mapPassthrough", "filter"]) {
     if (processArgs.containsKey(nam) && processArgs[nam]) {
-      assert processArgs[nam] instanceof Closure : "Expected process argument '$nam' to be null or a Closure. Found: class ${processArgs[nam].getClass()}"
+      assert processArgs[nam] instanceof Closure : "Error in module '$key': Expected process argument '$nam' to be null or a Closure. Found: class ${processArgs[nam].getClass()}"
+    }
+  }
+
+  // check fromState
+  assert processArgs.containsKey("fromState") : "Error in module '$key': fromState is a required argument"
+  def fromState = processArgs["fromState"]
+  assert fromState == null || fromState instanceof Closure || fromState instanceof Map || fromState instanceof List :
+    "Error in module '$key': Expected process argument 'fromState' to be null, a Closure, a Map, or a List. Found: class ${fromState.getClass()}"
+  if (fromState) {
+    // if fromState is a List, convert to map
+    if (fromState instanceof List) {
+      // check whether fromstate is a list[string]
+      assert fromState.every{it instanceof CharSequence} : "Error in module '$key': fromState is a List, but not all elements are Strings"
+      fromState = fromState.collectEntries{[it, it]}
+    }
+
+    // if fromState is a map, convert to closure
+    if (fromState instanceof Map) {
+      // check whether fromstate is a map[string, string]
+      assert fromState.values().every{it instanceof CharSequence} : "Error in module '$key': fromState is a Map, but not all values are Strings"
+      assert fromState.keySet().every{it instanceof CharSequence} : "Error in module '$key': fromState is a Map, but not all keys are Strings"
+      def fromStateMap = fromState.clone()
+      fromState = { it ->
```

Comment:
For some reason it took me a while to realise that this was actually defining the closure instead of iterating over the `fromStateMap` entries (might just be me). Could you add the comment like? `// define the closure` or something

Reply:
```suggestion
      // turn the map into a closure to be used later on
      fromState = { it ->
```